pkg/openstack/telemetry.go

```diff
@@ -92,6 +141,88 @@ func ReconcileTelemetry(ctx context.Context, instance *corev1beta1.OpenStackCont
 		instance.Spec.Telemetry.Template.Autoscaling.Aodh.TLS.API.Internal.SecretName = endpointDetails.GetEndptCertSecret(service.EndpointInternal)
 	}
 
+	if telemetry.Status.Conditions.IsTrue(telemetryv1.MetricStorageReadyCondition) {
+		// EnsureEndpoint for prometheus
+		// NOTE: We don't manage the prometheus service, it's managed by COO, we just annotate it
+		//       We don't have overrides.
+		endpointDetails, ctrlResult, err := EnsureEndpointConfig(
+			ctx,
+			instance,
+			helper,
+			telemetry,
+			&corev1.ServiceList{
+				Items: []corev1.Service{*prometheusSvc},
+			},
+			nil,
+			corev1beta1.Override{},
```

Comment:
I have a question about overrides. We don't have overrides for Ceilometer and MetricStorage. Is this something we should be adding? I'm not sure what it actually overrides.

Ceilometer doesn't have an API, it just exposes an sg-core endpoint with metrics, that are scraped by prometheus.
In MetricStorage we have the Prometheus and AlertManager endpoints, but the services are created by COO. We can annotate them, but we can't modify the rest of them.

Reply:
For the Ceilometer there's no need to add the overrides. For the MetricStorage I'm not entirely sure, but I don't think it's needed as well, if they're created by the COO. @olliewalsh is returning from PTO tomorrow, I'll ping him tomorrow if he could shed more light.

Apart from that it's good to go. Thank you very much for taking care of the telemetry!

Reply:
It's overrides for the public ingress Route(s) to e.g use a specific TLS cert from an external CA instead of our internal self-signed CA. If there is no route then doesn't apply.

Reply:
can see a route is created later though